### PR TITLE
Add infrastructure for Mozillians

### PR DESCRIPTION
**Pull request by flamingspaz**
- Production slaves
- Production DB
- Staging ES/EC

### CODE REVIEW

File: db.tf [hunk 1]
```diff
@@ -124,3 +134,45 @@ resource "aws_route53_record" "postgres-shared-dns" {
...
+    source_security_group_id = "${aws_security_group.mozillians-slave-ec2-sg.id}"
+
+    security_group_id = "${aws_security_group.mozillians-prod-rds-sg.id}"
+}
+
+resource "aws_db_instance" "mysql-mozillians-db" {
+    allocated_storage    = 40
+    engine               = "mysql"
+    engine_version       = "5.6.27"
+    instance_class       = "db.t2.medium"
```
Comment (johngian) [inline db.tf]: We can scale down here, given that we have a read replica.

Comment (johngian) [inline db.tf]: That said, where is the replica?

File: modules/mozillians/main.tf [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+"Version": "2012-10-17",
+"Statement": [
+    {
+        "Action": "es:*",
+        "Principal": "*",
+        "Effect": "Allow",
+        "Condition": {
+          "IpAddress": {
+            "aws:SourceIp": [
+              "52.0.0.42"
```
Comment (johngian) [inline modules/mozillians/main.tf]: How do you know that this is the source ip?

File: modules/mozillians/main.tf [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+resource "aws_elasticsearch_domain" "mozillians-metrics-es" {
+    domain_name                       = "mozillians-metrics-es"
+    elasticsearch_version             = "2.3"
+
+    access_policies                   = "${var.mozillians-es-iam-policy}"
+
+    ebs_options {
+        ebs_enabled                   = true
+        volume_type                   = "standard"
+        volume_size                   = 50
```
Comment (johngian) [inline modules/mozillians/main.tf]: We need waaaay less.

File: modules/mozillians/main.tf [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    cluster_config {
+        instance_count                = 2
```
Comment (johngian) [inline modules/mozillians/main.tf]: This should be 3.

Comment (johngian) [inline modules/mozillians/main.tf]: Also please make sure it lives in the same region with the slaves.

File: modules/mozillians/main.tf (hunk 3 above)
Comment (johngian) [inline modules/mozillians/main.tf]: The minimum (10G) should be more than fine.

File: mozillians.tf [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+resource "aws_launch_configuration" "mozillians-slave-ec2-lc" {
+  name_prefix                 = "mozillians-slave-production-lc"
+  image_id                    = "${lookup(var.aws_amis, var.aws_region)}"
+  instance_type               = "t2.medium"
+  key_name                    = "ansible"
+  security_groups             = ["${aws_security_group.mozillians-slave-ec2-sg.id}"]
+  associate_public_ip_address = true
+  root_block_device {
+    volume_size               = 100
```
Comment (johngian) [inline mozillians.tf]: Since only mozillians.org is going to live in this node, i would go for less.

File: mozillians.tf [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      value                 = "mozillians"
+      propagate_at_launch   = true
+    }
+
+    lifecycle {
+      create_before_destroy = true
+    }
+}
+
+module "mozillians-staging" {
```
Comment (johngian) [inline mozillians.tf]: Do we need to separate the staging nodes from prod? I think having mozillians.org nodes for all the envs should be fine.

File: modules/mozillians/main.tf (hunk 2 above)
Comment (flamingspaz) [inline modules/mozillians/main.tf]: We don't, It's a placeholder until the prod slaves are up. Unfortunately Terraform's `aws_instance` data source can only match to a single instance so we can't use it to populate this.

Comment (johngian) [inline modules/mozillians/main.tf]: Can't we put an ELB in front?

Comment (johngian) [inline modules/mozillians/main.tf]: The slave IPs are not something to rely on

File: mozillians.tf (hunk 6 above)
Comment (flamingspaz) [inline mozillians.tf]: This puts up the supporting infrastructure like Elasticache and ES which we'll have two of. That said we decided to use one ES cluster so I'll move that out.

File: mozillians.tf (hunk 5 above)
Comment (flamingspaz) [inline mozillians.tf]: does 20GB sound reasonable?

Comment (johngian) [inline mozillians.tf]: yeah i think so

**Comment by flamingspaz** [comment]
@johngian this is ready to be re-reviewed

File: db.tf [hunk 7]
```diff
@@ -10,6 +11,15 @@ resource "aws_db_subnet_group" "apps-shared-rds-subnetgroup" {
     }
 }
 
+resource "aws_db_subnet_group" "apps-production-rds-subnetgroup" {
+    name = "apps-production-rds-subnetgroup"
+    description = "RDS subnet group for production VPC"
+    subnet_ids = ["${aws_subnet.apps-production-1a.id}", "${aws_subnet.apps-production-1c.id}", "${aws_subnet.apps-production-1d.id}"]
```
Comment (johngian) [inline db.tf]: split to multiple lines here.

File: mozillians.tf [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    type              = "ingress"
+    from_port         = 0
+    to_port           = 0
+    protocol          = "-1"
+    cidr_blocks       = ["${aws_vpc.apps-shared-vpc.cidr_block}"]
+
+    security_group_id = "${aws_security_group.mozillians-slave-ec2-sg.id}"
+}
+
+resource "aws_launch_configuration" "mozillians-slave-ec2-lc" {
```
Comment (johngian) [inline mozillians.tf]: Please add the profile used in the rest of launch configurations

File: mozillians.tf [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+resource "aws_elasticsearch_domain" "mozillians-es" {
+    domain_name                       = "mozillians-shared-es"
+    elasticsearch_version             = "2.3"
+
+    ebs_options {
+        ebs_enabled                   = true
+        volume_type                   = "standard"
+        volume_size                   = 10
```
Comment (johngian) [inline mozillians.tf]: this means 3x10 G. I suggest we go with 1 and scale in the future.

Comment (johngian) [inline mozillians.tf]: 3x1G should be more than enough

Comment (flamingspaz) [inline mozillians.tf]: Minimum for ES is 10GB

**Pull request merged**